backend_test.go

```diff
@@ -76,70 +94,95 @@ func TestBackendsNilKeys(t *testing.T) {
 				// Nil keys are treated as the empty key for most operations.
 				expect := func(key, value []byte) {
 					if len(key) == 0 { // nil or empty
-						assert.Equal(t, db.Get(nil), db.Get([]byte("")))
+						nilValue, err := db.Get(nil)
+						assert.NoError(t, err)
+						byteValue, err := db.Get([]byte(""))
+						assert.NoError(t, err)
+						assert.Equal(t, nilValue, byteValue)
 						assert.Equal(t, db.Has(nil), db.Has([]byte("")))
 					}
-					assert.Equal(t, db.Get(key), value)
+					value2, err := db.Get(key)
+					assert.NoError(t, err)
+					assert.Equal(t, value2, value)
 					assert.Equal(t, db.Has(key), value != nil)
 				}
 
 				// Not set
 				expect(nil, nil)
 
 				// Set nil value
-				db.Set(nil, nil)
+				err := db.Set(nil, nil)
+				require.NoError(t, err)
 				expect(nil, []byte(""))
 
 				// Set empty value
-				db.Set(nil, []byte(""))
+				err = db.Set(nil, []byte(""))
+				require.NoError(t, err)
 				expect(nil, []byte(""))
 
 				// Set nil, Delete nil
-				db.Set(nil, []byte("abc"))
+				err = db.Set(nil, []byte("abc"))
+				require.NoError(t, err)
 				expect(nil, []byte("abc"))
-				db.Delete(nil)
+				err = db.Delete(nil)
+				require.NoError(t, err)
 				expect(nil, nil)
 
 				// Set nil, Delete empty
-				db.Set(nil, []byte("abc"))
+				err = db.Set(nil, []byte("abc"))
+				require.NoError(t, err)
 				expect(nil, []byte("abc"))
```

Comment:
Do we still need these `expect`s? I think those were there to catch the panic no?